core/src/avm2/activation.rs

```diff
@@ -2463,6 +2478,254 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
+
+        Ok(FrameControl::Continue)
+    }
+
+    /// Implements `Op::Sf32`
+    fn op_sf32(&mut self) -> Result<FrameControl<'gc>, Error> {
+        let address = self.context.avm2.pop().coerce_to_i32(self)?;
+        let val = self.context.avm2.pop().coerce_to_number(self)? as f32;
+
+        let dm = self.domain_memory()?;
+
+        if address < 0 || (address as usize + 3) >= dm.read().len() {
+            return Err("RangeError: The specified range is invalid".into());
+        }
+
+        dm.write(self.context.gc_context)
+            .write_bytes_at(&val.to_le_bytes(), address as usize);
+
+        Ok(FrameControl::Continue)
+    }
+
+    /// Implements `Op::Sf64`
+    fn op_sf64(&mut self) -> Result<FrameControl<'gc>, Error> {
+        let address = self.context.avm2.pop().coerce_to_i32(self)?;
+        let val = self.context.avm2.pop().coerce_to_number(self)?;
+
+        let dm = self.domain_memory()?;
+
+        if address < 0 || (address as usize + 7) >= dm.read().len() {
+            return Err("RangeError: The specified range is invalid".into());
+        }
+
+        dm.write(self.context.gc_context)
+            .write_bytes_at(&val.to_le_bytes(), address as usize);
+
+        Ok(FrameControl::Continue)
+    }
+
+    /// Implements `Op::Li8`
+    fn op_li8(&mut self) -> Result<FrameControl<'gc>, Error> {
+        let address = self.context.avm2.pop().coerce_to_u32(self)? as usize;
+
+        let dm = self.domain_memory()?;
+        let val = dm.read().get(address);
+
+        if let Some(val) = val {
+            self.context.avm2.push(Value::Integer(val as i32));
+        } else {
+            return Err("RangeError: The specified range is invalid".into());
+        }
+
+        Ok(FrameControl::Continue)
+    }
+
+    /// Implements `Op::Li16`
+    fn op_li16(&mut self) -> Result<FrameControl<'gc>, Error> {
+        let address = self.context.avm2.pop().coerce_to_u32(self)? as usize;
+
+        let dm = self.domain_memory()?;
+        let r = dm.read();
+        let val = r.get_range(address..address + 2);
+
+        if let Some(val) = val {
+            let buf: [u8; 4] = [val[0], val[1], 0, 0];
+            self.context
+                .avm2
+                .push(Value::Integer(i32::from_le_bytes(buf)));
+        } else {
+            return Err("RangeError: The specified range is invalid".into());
+        }
+
+        Ok(FrameControl::Continue)
+    }
+
+    /// Implements `Op::Li32`
+    fn op_li32(&mut self) -> Result<FrameControl<'gc>, Error> {
+        let address = self.context.avm2.pop().coerce_to_u32(self)? as usize;
+
+        let dm = self.domain_memory()?;
+        let r = dm.read();
+        let val = r.get_range(address..address + 4);
+
+        if let Some(val) = val {
+            let buf: [u8; 4] = [val[0], val[1], val[2], val[3]];
+
+            self.context
+                .avm2
+                .push(Value::Integer(i32::from_le_bytes(buf)));
```

Comment:
I think I'd prefer to do:
```
i32::from_le_bytes(val.try_into().unwrap())
```
and ditto for all the other ops. Checked on Godbolt and this seems to get optimized properly, too.